### PR TITLE
OBJ-240 Chips rest interface client

### PR DESCRIPTION
**Pull request by ICoxCH**

### CODE REVIEW

File: src/main/java/uk/gov/companieshouse/api/strikeoffobjections/chips/ChipsClient.java [hunk 1]
```diff
@@ -1,23 +1,37 @@
...
     public void sendToChips(String requestId, ChipsRequest chipsRequest){
-        // TODO OBJ-240 add contact model object and send to CHIPS
         apiLogger.infoContext(requestId,
                 String.format("Posting %s to chips rest interfaces", chipsRequest));
+
+        ResponseEntity<String> chipsRestResponse = restTemplate.postForEntity(chipsRestUrl, chipsRequest, String.class);
+
+        apiLogger.infoContext(
+                requestId,
+                String.format("Sent data to CHIPS, received status code: %s", chipsRestResponse.getStatusCode())
```
Comment (markpit) [inline src/main/java/uk/gov/companieshouse/api/strikeoffobjections/chips/ChipsClient.java:34]: would it be worth including the url in the log message as well? might help us debug when we have to look through live logs.

Comment (ICoxCH) [reply src/main/java/uk/gov/companieshouse/api/strikeoffobjections/chips/ChipsClient.java:43]: Added to the log message before the data is sent, as if the request fails this log may not be executed

**Review by markpit — APPROVED** [review]

File: src/main/java/uk/gov/companieshouse/api/strikeoffobjections/chips/ChipsClient.java [hunk 2]
```diff
@@ -1,23 +1,46 @@
...
     public void sendToChips(String requestId, ChipsRequest chipsRequest){
-        // TODO OBJ-240 add contact model object and send to CHIPS
-        apiLogger.infoContext(requestId,
-                String.format("Posting %s to chips rest interfaces", chipsRequest));
+        Map<String, Object> logMap = new HashMap<>();
+        logMap.put("chipsRestUrl", chipsRestUrl);
+
+        apiLogger.infoContext(
+                requestId,
+                String.format("Posting %s to chips rest interfaces", chipsRequest),
```
Comment (mwestacott) [inline src/main/java/uk/gov/companieshouse/api/strikeoffobjections/chips/ChipsClient.java:35]: So minor, but 'CHIPS' in uppercase to match how you've written it in log below.

**Review by mwestacott — APPROVED** [review]
Minor observation on log message.

**Review by mwestacott — APPROVED** [review]

**Pull request merged**